src/model/SequenceView.tsx

```diff
@@ -0,0 +1,222 @@
+      <Sequence sequence={dataToDisplay.value} />
+    </Fragment>
+  );
+};
+
+export const IsoformInfo: React.FC<{ isoformData: Isoform }> = ({
+  isoformData,
+}) => {
+  const infoListData = [
+    {
+      title: 'Name',
+      content: isoformData.name.value,
+    },
+    {
+      title: 'Synonyms',
+      content:
+        isoformData.synonyms &&
+        isoformData.synonyms.map(syn => syn.value).join(', '),
+    },
+    {
+      title: 'Note',
+      content:
+        isoformData.note &&
+        isoformData.note.texts.map(note => note.value).join(', '),
+    },
+  ];
+  // TODO isoformData.sequenceIds is used to get the features for
+  // splice variants - they need to be somehow displayed
+  return (
+    <Fragment key={isoformData.isoformIds.join('')}>
+      <hr />
+      <h4>{isoformData.isoformIds.join(', ')}</h4>
+      {isoformData.isoformSequenceStatus === 'Displayed' && (
+        <p>
+          This isoform has been chosen as the <strong>canonical</strong>{' '}
+          sequence. All positional information in this entry refers to it. This
+          is also the sequence that appears in the downloadable versions of the
+          entry.
+        </p>
+      )}
+      <InfoList infoData={infoListData} />
+    </Fragment>
+  );
+};
+
+export const SequenceViewEntry: React.FC<SequenceViewProps> = ({ data }) => {
+  const flag = idx(data, _ => _.proteinDescription.flag);
+  const sequenceInfoData = [
+    {
+      title: 'Sequence status',
+      content:
+        flag &&
+        [
+          Flag.FRAGMENT,
+          Flag.FRAGMENTS,
+          Flag.FRAGMENTS_PRECURSOR,
+          Flag.FRAGMENT_PRECURSOR,
+        ].includes(flag)
+          ? flag
+          : 'Complete',
+    },
+    {
+      title: 'Sequence processing',
+      content:
+        flag &&
+        [
+          Flag.PRECURSOR,
+          Flag.FRAGMENTS_PRECURSOR,
+          Flag.FRAGMENT_PRECURSOR,
+        ].includes(flag) &&
+        'The displayed sequence is further processed into a mature form.',
+    },
+  ];
+
+  if (!data.comments) {
+    return null;
+  }
+  const alternativeProducts = data.comments.find(
+    comment => comment.commentType === FreeTextType.ALTERNATIVE_PRODUCTS
+  );
+  if (!alternativeProducts) {
+    return null;
+  }
+  return (
+    <Fragment>
+      <InfoList infoData={sequenceInfoData} />
+      <p>
+        This entry describes{' '}
+        <strong>{alternativeProducts.isoforms.length}</strong> isoforms produced
+        by <strong>{alternativeProducts.events.join(' & ')}</strong>.
+      </p>
+      <p>{alternativeProducts.note.texts.map(text => text.value).join(' ')}</p>
+      {alternativeProducts.isoforms.map(isoform => {
+        let isoformSequence, lastUpdateDate;
+        // Canonical
+        if (isoform.isoformSequenceStatus === 'Displayed') {
+          isoformSequence = data.sequence;
+          if (data.entryAudit) {
+            lastUpdateDate = `${data.entryAudit.lastSequenceUpdateDate} v${
```

Comment:
Is all of `data.entryAudit.lastSequenceUpdateDate` guaranteed to exist?

Reply:
It is if `entryAudit` is present. I've changed the type def to reflect this.